File: ruby-pr/53/ruby/code-quality-logging/secure_profile_updater.rb
class SecureProfileUpdater
  def update_profile(user_id, profile_data)
    # {fact rule=code-quality-logging@v1.0 defects=0}
    puts "Updating profile for user #{user_id}"
    
    allowed_fields = %w[name address city]
    log_data = profile_data.select { |k, v| allowed_fields.include?(k.to_s) }
    puts "Updated fields: #{log_data.keys}"
    
    # Update logic
    return true
    # {/fact}
  end
end

File: php-pr/51/php/code-quality-logging/SecureProfileUpdater.php
<?php

class SecureProfileUpdater
{
    public function updateProfile($userId, $profileData)
    {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        error_log("Updating profile for user {$userId}");
        
        $allowedFields = ['name', 'address', 'city'];
        $logData = array_intersect_key($profileData, array_flip($allowedFields));
        error_log("Updated fields: " . json_encode(array_keys($logData)));
        
        // Update logic
        return true;
        // {/fact}
    }
}